### PR TITLE
ref(grouping): Rewrite hierarchical grouping tests

### PR DESCRIPTION
**Pull request by armenzg**
This makes the tests a little easier to understand and prepares for my next set of changes.

### CODE REVIEW

File: tests/sentry/event_manager/test_hierarchical_hashes.py [hunk 1]
```diff
@@ -202,54 +202,18 @@ def test_partial_move(default_project, fast_save):
 
 
 class EventManagerGroupingTest(TestCase):
-    def test_applies_secondary_grouping_hierarchical(self):
```
Comment (armenzg) [inline tests/sentry/event_manager/test_hierarchical_hashes.py:205]: This is easier to review with the split view
image

File: tests/sentry/event_manager/test_hierarchical_hashes.py [hunk 2]
```diff
@@ -202,54 +202,18 @@ def test_partial_move(default_project, fast_save):
...
-        project = self.project
-        project.update_option("sentry:grouping_config", "legacy:2019-03-12")
-        project.update_option("sentry:secondary_grouping_expiry", 0)
```
Comment (armenzg) [inline tests/sentry/event_manager/test_hierarchical_hashes.py:208]: These 3 lines now live in the shared function `set_options`:
image

File: tests/sentry/event_manager/test_hierarchical_hashes.py [hunk 3]
```diff
@@ -202,54 +202,18 @@ def test_partial_move(default_project, fast_save):
...
-
-        timestamp = time.time() - 300
```
Comment (armenzg) [inline tests/sentry/event_manager/test_hierarchical_hashes.py:210]: `save_event` will not use timestamp anymore.

File: tests/sentry/event_manager/test_hierarchical_hashes.py [hunk 4]
```diff
@@ -202,54 +202,18 @@ def test_partial_move(default_project, fast_save):
...
-
-        def save_event(ts_offset):
```
Comment (armenzg) [inline tests/sentry/event_manager/test_hierarchical_hashes.py:212]: This function has been converted into a helper function. See down below.

File: tests/sentry/event_manager/test_hierarchical_hashes.py [hunk 5]
```diff
@@ -202,54 +202,18 @@ def test_partial_move(default_project, fast_save):
...
-                        ]
-                    },
-                )
-            )
-            manager.normalize()
-            with self.tasks():
-                return manager.save(project.id)
-
-        event = save_event(0)
+    def test_can_upgrade_to_hierarchical_config(self):
```
Comment (armenzg) [inline tests/sentry/event_manager/test_hierarchical_hashes.py:205]: More meaningful naming.

File: tests/sentry/event_manager/test_hierarchical_hashes.py [hunk 6]
```diff
@@ -202,54 +202,18 @@ def test_partial_move(default_project, fast_save):
...
+        self.set_options("legacy:2019-03-12")  # Starting configuration
 
-        project.update_option("sentry:grouping_config", "mobile:2021-02-12")
-        project.update_option("sentry:secondary_grouping_config", "legacy:2019-03-12")
-        project.update_option("sentry:secondary_grouping_expiry", time.time() + (24 * 90 * 3600))
```
Comment (armenzg) [inline tests/sentry/event_manager/test_hierarchical_hashes.py:246]: These 3 lines have been moved into a helper function.
image

File: tests/sentry/event_manager/test_hierarchical_hashes.py [hunk 7]
```diff
@@ -325,6 +255,42 @@ def save_event(ts_offset):
...
-        event3 = save_event(4)
+        self.project.update_option("sentry:secondary_grouping_expiry", 0)
+        event3 = self.save_event()
         assert event3.group_id == event2.group_id
+
+    def save_event(self):
+        manager = EventManager(
+            make_event(
+                message="foo 123",
+                event_id=hex(2**127)[-32:],
```
Comment (armenzg) [inline tests/sentry/event_manager/test_hierarchical_hashes.py:266]: Using the timestamp offset was not necessary. Two events can still have the same ID.
image

**Review by snigdhas — APPROVED** [review]

**Pull request merged**